src/NServiceBus.Hosting.ComponentTests/When_starting_and_stopping_host.cs

```diff
@@ -0,0 +1,134 @@
+﻿namespace NServiceBus.Hosting.ComponentTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class When_starting_and_stopping_host
+    {
+        [Test]
+        public void Start_and_stop_are_called_for_scanned_type()
+        {
+            var context = RunHost();
+
+            Assert.True(context.ScannedStartCalled);
+            Assert.True(context.ScannedStopCalled);
+        }
+
+        [Test]
+        public void Start_and_stop_are_called_for_instance()
+        {
+            var context = RunHost();
+
+            Assert.True(context.InstanceStartCalled);
+            Assert.True(context.InstanceStopCalled);
+        }
+
+        [Test]
+        public void Instance_is_registered_once()
+        {
+            var context = RunHost();
+
+            Assert.True(context.InstanceTimesRegistered == 1);
```

Comment:
I added this test because I noticed that there was nothing excluding the registered instances from assembly scanning, so this is current behavior but it seems counter intuitive.

I don't know if there are any users that might be accidentally relying on this behavior?

Thoughts?